### PR TITLE
Add test for online extending lvm volume in cinder

### PR DESCRIPTION
**Pull request by samuelallan72**
Adding a test to cover https://bugs.launchpad.net/charm-cinder/+bug/1939389 | https://review.opendev.org/c/openstack/charm-cinder/+/842902

### CODE REVIEW

File: zaza/openstack/charm_tests/cinder/tests.py [hunk 1]
```diff
@@ -214,6 +214,32 @@ def test_120_volume_force_delete(self):
...
+
+        # check we have a LVM backed image
+        host = getattr(test_vol, 'os-vol-host-attr:host').split('#')[0]
+        self.assertIn('@LVM', host)
+
+        instance = self.launch_guest('zaza-cinder-200')
+        openstack_utils.attach_volume(self.nova_client, test_vol.id, instance.id)
+        test_vol.extend('UNUSED_ARG_XXX', '2')
+
+        # TODO: assert volume is resized successfully
```
[the remark this answers is not in the document]
Comment (wolsen) [reply zaza/openstack/charm_tests/cinder/tests.py:244]: As a thought, you'd probably check the new size of the volume by listing the volume again.

File: zaza/openstack/charm_tests/cinder/tests.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ def setUpClass(cls):
         cls.lead_unit = zaza.model.get_lead_unit_name(
             "cinder", model_name=cls.model_name)
         cls.cinder_client = openstack_utils.get_cinder_session_client(
-            cls.keystone_session)
+            # cls.keystone_session)
+            cls.keystone_session, version=3.42)
```
[the remark this answers is not in the document]
Comment (nobuto-m) [reply zaza/openstack/charm_tests/cinder/tests.py:48]: The micro version is based on the following description.
https://docs.openstack.org/cinder/latest/cli/cli-manage-volumes.html#extend-attached-volume
> Starting from microversion 3.42, it is also possible to extend an attached volume with status in-use, depending upon policy settings and the capabilities of the backend storage.

I'm not sure what's the best documentation describing the micro version scheme, but I believe a newer micro version is opt-in when poking API endpoint.
https://docs.openstack.org/api-guide/compute/microversions.html

File: zaza/openstack/charm_tests/cinder/tests.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nobuto-m) [reply zaza/openstack/charm_tests/cinder/tests.py]: Maybe, logging in to a VM by SSH and check `lsblk` or something? I thought there was a SSH command wrapper in Zaza somewhere.

File: zaza/openstack/charm_tests/cinder/tests.py [hunk 3]
```diff
@@ -214,6 +242,83 @@ def test_120_volume_force_delete(self):
...
+        def verify(stdin, stdout, stderr):
+            status = stdout.channel.recv_exit_status()
+            self.assertEqual(status, 0)
+            output = stdout.read().decode()
+            print(output)
+            self.assertIn('2G', output)
+
+        openstack_utils.ssh_command(
+            username, fip, instance_name,
+            'lsblk',
```
Comment (nobuto-m) [inline zaza/openstack/charm_tests/cinder/tests.py:319]: I checked this with Cirros. Hope this helps.

```
$ lsblk -sn -o SIZE /dev/vda 
  6G
$ lsblk -snb -o SIZE /dev/vda 
6442450944
```

**Review by ajkavanagh — APPROVED** [review]

**Pull request merged**